crates/bin/pcli/src/command/tx.rs

```diff
@@ -1280,6 +1335,105 @@ impl TxCmd {
+                let view: &mut dyn ViewClient = app
+                    .view
+                    .as_mut()
+                    .context("view service must be initialized")?;
+                let start_height = view.status().await?.full_sync_height;
+
+                // TODO: make this a valid calculation.
+                let end_height = start_height + auction_duration;
+
+                let _block_interval = end_height.checked_sub(start_height).ok_or_else(|| {
+                    anyhow::anyhow!(
+                        "block interval calculation has underflowed (end={}, start={})",
+                        end_height,
+                        start_height
+                    )
+                })?;
+
+                // TODO: what should the step_count be?
+
+                let _starting_price_ = starting_price.parse::<Value>()?;
+                let _reserve_price = reserve_price.parse::<Value>()?;
+
+                let mut planner = Planner::new(OsRng);
+                planner
+                    .set_gas_prices(gas_prices)
+                    .set_fee_tier((*fee_tier).into());
+
+                // TODO: add `ActionDutchAuctionSchedule` to planner (depends on #4248).
+
+                let plan = planner
+                    .plan(
+                        app.view
+                            .as_mut()
+                            .context("view service must be initialized")?,
+                        AddressIndex::new(*source),
+                    )
+                    .await
+                    .context("can't build send transaction")?;
+                app.build_and_submit_transaction(plan).await?;
+            }
+            TxCmd::DutchAuctionWithdraw {
+                auction_id: _,
+                source,
+                fee_tier,
+            } => {
+                let mut planner = Planner::new(OsRng);
+                planner
+                    .set_gas_prices(gas_prices)
+                    .set_fee_tier((*fee_tier).into());
+
+                // TODO: add `ActionDutchAuctionWithdraw` to planner (depends on #4248).
+
+                let plan = planner
+                    .plan(
+                        app.view
+                            .as_mut()
+                            .context("view service must be initialized")?,
+                        AddressIndex::new(*source),
+                    )
+                    .await
+                    .context("can't build send transaction")?;
+                app.build_and_submit_transaction(plan).await?;
+            }
+            TxCmd::DutchAuctionEnd {
+                auction_id: _,
+                source,
+                fee_tier,
+            } => {
+                let mut planner = Planner::new(OsRng);
+                planner
+                    .set_gas_prices(gas_prices)
+                    .set_fee_tier((*fee_tier).into());
+
+                // TODO: add `ActionDutchAuctionEnd` to planner (depends on #4248).
```

Comment:
FWIW, we can do away with the boilerplate and make those branches `todo!()`, your call.

Reply:
what if we merge #4248 first?

Reply:
I think the only boilerplate we should keep is the planner related code since we'll be passing all the necessary CLI parameters anyways.

Reply:
Sounds good